en/debug/porting-guide.md

```diff
@@ -1,25 +1,51 @@
-# Porting Guide
+# Flight Controller Porting Guide
 
-PX4 consists of two main layers: The board support and middleware layer on top of the host OS \(NuttX, Linux or any other POSIX platform like Mac OS\). And the applications \(in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
+This topic for developers who want to port PX4 to work with *new* flight controller hardware.
+
+## Architecture
+
+PX4 consists of two main layers: The board support and middleware layer on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS). And the applications (in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
 
 This guide is focused only on the middleware as the applications will run on any board target.
 
 ## NuttX Boards
 
-The main files for a NuttX board are located at \(for the FMUv5 example\):
+The location of the main files for a NuttX board are:
+
+* Board startup and PX4 board configuration: [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards). It contains bus and GPIO mappings and the board initialization code. 
+  * FMUv5 example: [src/drivers/boards/px4fmu-v5](https://github.com/PX4/Firmware/tree/master/src/drivers/boards/px4fmu-v5). 
+* NuttX board config: [nuttx-configs](https://github.com/PX4/Firmware/tree/master/nuttx-configs). The OS gets loaded as part of the application build. 
```

Comment:
What is the difference between NuttX board config and NuttX OS config.

Reply:
@dagar What is the difference between NuttX board config and NuttX OS config?

Reply:
I'm not sure I have a great answer for this. The nuttx-configs define the actual nuttx os (defconfig). The board driver within px4 handles board init, i2c, spi, etc, and would typically also live within nuttx-config (for a regular nuttx board), but in our case also depend on PX4 components (hrt driver, param init, etc) that don't belong in nuttx.

Reply:
Me either :-) Also not sure of the value of saying "The OS gets loaded as part of the application build".

I think that you are saying that "NuttX OS config" is pure OS configuration (independent of PX4), while  the "NuttX board config" contains PX4 frameworks for integrating with the specific PX4 board - essentially the PX4 components below the hardware abstraction layer that work with NuttX. So for example this would contain the driver framework for spi, but the actual drivers would be elsewhere. Could be completely wrong!

If so, how about I add a sentence to differentiate (in italic below):

* NuttX board config: [nuttx-configs](https://github.com/PX4/Firmware/tree/master/nuttx-configs). *This includes PX4 components and frameworks for integrating with NuttX (e.g. driver layers etc.).*